src/pages/user-descriptor/line-graph.jsx

```diff
@@ -63,216 +49,225 @@ sortedTimeArray.sort(function(a, b) {
   b = b.split("-");
   return new Date(a[1], a[0], 1) - new Date(b[1], b[0], 1);
 });
-
-// dataArray.sort(compareValues('weight'));
-let weightArray = [];
-for (let j = 0; j < dataArray.length; j++) {
-  weightArray[j] = dataArray[j].weight;
-}
-
-// temporary dataset used to populate graph
-let rangeLength = 11;
-let rangearraym = new Array(rangeLength);
-rangearraym.fill(0);
-let rangearrayf = new Array(rangeLength);
-rangearrayf.fill(0);
-
-// finding range of users weight
-function weightRange(WTObj, weightRangeArray) {
-  let indx;
-  if (WTObj.weight >= 40 && WTObj.weight <= 50) {
-    weightRangeArray[0] = weightRangeArray[0] + 1;
-    indx = WTObj.weight;
-  } else if (WTObj.weight >= 51 && WTObj.weight <= 100) {
-    weightRangeArray[1] = weightRangeArray[1] + 1;
-    indx = WTObj.weight;
-  } else if (WTObj.weight >= 101 && WTObj.weight <= 150) {
-    weightRangeArray[2] = weightRangeArray[2] + 1;
-    indx = WTObj.weight;
-  } else if (WTObj.weight >= 151 && WTObj.weight <= 200) {
-    weightRangeArray[3] = weightRangeArray[3] + 1;
-    indx = WTObj.weight;
-  } else if (WTObj.weight >= 201 && WTObj.weight <= 300) {
-    weightRangeArray[4] = weightRangeArray[4] + 1;
-    indx = WTObj.weight;
-  } else if (WTObj.weight >= 301) {
-    weightRangeArray[5] = weightRangeArray[5] + 1;
-    indx = WTObj.weight;
-  } else {
-    indx = 0;
+export function LineGraph() {
+  let timelist = [];
+  let obj = DataFunction();
+  let count = 4;
+  for (let i = 0; i < obj.length; i++) {
+    let unixTime = obj[i].unixTimestamp;
+    let startDate = new Date(unixTime + 604000000 * count);
+    let month = startDate.getMonth(); //Be careful! January is 0 not 1
+    let year = startDate.getFullYear();
+    let dateString = month + 1 + "-" + year;
+    timelist[i] = dateString;
+    count = count + 4;
   }
-  return indx;
-}
-
-let usergender;
-let weightIndx; // used to determine users weight index
-
-// record user info
-let weightIndex;
-let userValue;
-for (let i = 0; i < dataArray.length; i++) {
-  if (dataArray[i].gender == "Male") {
-    weightIndex = weightRange(dataArray[i], rangearraym);
-  } else if (dataArray[i].gender == "Female") {
-    weightIndex = weightRange(dataArray[i], rangearrayf);
+  timelist.sort(function(a, b) {
```

Comment:
Don't sort data, it should be presorted from the server. So make the mockserver return sorted dates if necessary. If it's not sorted, its bad data and we will fix in backend.

Reply:
since the mock server return same date, I created mock data keeping account the starting date received from the server and then incrementing by month and sorted it. If it needs to be changed, changes need to be made in the time generation code in resolver.